### PR TITLE
Standards and styles : std usage, auto, nullptr, rtti

### PR DESCRIPTION
**Pull request by michalpelka**

### CODE REVIEW

**Comment by michalpelka** [comment]
Here I went through the code base and covered:
#310 and RTTI task from #311
# Use AZ or AZStd
Changed some std to AZStd, please note that some portion of the codebase (also in ROS.Static) links a lot from the standard library through `rclcpp`.
# Use of auto
Adjusted usage of `auto` to match [guideline](https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#use-of-auto)
# Use of nullptr
No problem here
# Exceptions and RTTI
I disabled bort RTTI and exception from ROS2.Static/ROS2 targets. 
Note that still there is need for RTTI and exceptions in ROS2.Editor. 
It is caused mainly by URDF parse library. It could be solved by incorporating that library into codebase - disabling `std::runtime_error`

File: Code/Source/Camera/CameraSensor.cpp [hunk 1]
```diff
@@ -32,13 +30,19 @@ namespace ROS2
...
+            { AZ::RHI::Format::R32_FLOAT, "32FC1" },
+        };
 
+        AZStd::unordered_map<AZ::RHI::Format, int> BitDepth{
+            { AZ::RHI::Format::R8G8B8A8_UNORM, 4 * sizeof(uint8_t) },
+            { AZ::RHI::Format::R16G16B16A16_UNORM, 4 * sizeof(uint16_t) },
+            { AZ::RHI::Format::R32G32B32A32_FLOAT, 4 * sizeof(float) }, // Unsuported by RVIZ2
+            { AZ::RHI::Format::R8_UNORM, sizeof(uint8_t) },
+            { AZ::RHI::Format::R16_UNORM, sizeof(uint16_t) },
+            { AZ::RHI::Format::R32_FLOAT, sizeof(float) },
```
Comment (adamdbrw) [inline Code/Source/Camera/CameraSensor.cpp:49]: This change block seems to be unrelated to style changes

File: Code/Source/Camera/CameraSensor.cpp [hunk 2]
```diff
@@ -188,10 +192,10 @@ namespace ROS2
                 AZ_Assert(Internal::FormatMappings.contains(format), "Unknown format in result %u", static_cast<uint32_t>(format));
                 sensor_msgs::msg::Image message;
                 message.encoding = Internal::FormatMappings.at(format);
+
```
Comment (adamdbrw) [inline Code/Source/Camera/CameraSensor.cpp]: unnecessary newline

File: Code/Source/Camera/CameraSensor.cpp [hunk 3]
```diff
@@ -188,10 +192,10 @@ namespace ROS2
...
                 message.width = descriptor.m_size.m_width;
                 message.height = descriptor.m_size.m_height;
-                message.step = message.width * sensor_msgs::image_encodings::bitDepth(message.encoding) / 8 *
-                    sensor_msgs::image_encodings::numChannels(message.encoding);
+                message.step = message.width * Internal::BitDepth.at(format);
```
Comment (adamdbrw) [inline Code/Source/Camera/CameraSensor.cpp:201]: unrelated?

File: Code/Source/Camera/CameraSensor.cpp (hunk 1 above)
Comment (michalpelka) [reply Code/Source/Camera/CameraSensor.cpp:49]: I disagree - it is related to the elimination of the usage of exception, what is mentioned in [style](https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#c-exceptions). With this change I do not need to include `sensor_msgs/image_encodings.hpp` that uses exception in header's code

File: Code/Source/Camera/CameraSensor.cpp (hunk 3 above)
Comment (michalpelka) [reply Code/Source/Camera/CameraSensor.cpp:201]: I disagree - it is related to the elimination of the usage of exception, what is mentioned in [style](https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#c-exceptions). With this change I do not need to include `sensor_msgs/image_encodings.hpp` that uses exception in header's code

**Review by adamdbrw — APPROVED** [review]
